cmd/root.go

```diff
@@ -0,0 +1,275 @@
+		if err != nil {
+			os.Exit(1)
+		}
+
+		var selectedTemplate t.Template
+		for _, template := range templates {
+			if template.Name == choice.String {
+				selectedTemplate = template
+				break
+			}
+		}
+
+		fmt.Printf(
+			" : Downloading template assets from %s/%s...\n",
+			selectedTemplate.Owner, selectedTemplate.Repository,
+		)
+
+		tempDir, err := os.MkdirTemp("", "gmy-template-")
+		defer os.RemoveAll(tempDir)
+		if err != nil {
+			fmt.Println(" ! Failed to create temporary folder :(")
+			os.Exit(1)
+		}
+
+		downloadUrl, err := t.GetDownloadUrl(selectedTemplate)
+		if err != nil {
+			fmt.Println(" ! Failed to fetch template details :(")
+			os.Exit(1)
+		}
+
+		resp, err := grab.Get(tempDir, downloadUrl)
+		if err != nil {
+			fmt.Printf(" ! Failed to download assets from %s\n", downloadUrl)
+			os.Exit(1)
+		}
+		err = resp.Err()
+		if err != nil {
+			fmt.Printf(" ! Failed to download assets from %s\n", downloadUrl)
+			os.Exit(1)
+		}
+
+		fmt.Println(" + Assets downloaded")
+		fmt.Println(" : Setting up project folder...")
+
+		file, err := os.Open(resp.Filename)
+		if err != nil {
+			fmt.Println(" ! Failed to extract downloaded assets: ", err)
+			os.Exit(1)
+		}
+
+		err = files.Untar(projectName, file)
+		if err != nil {
+			fmt.Println(" ! Failed to extract downloaded assets: ", err)
+			os.Exit(1)
+		}
+
+		fmt.Printf(`
+ + Project folder set up successfully!
+   Additional configuration options for you:
+
+`)
+
+		gitPrompt := confirmation.New(" ? Initialize Git repository", confirmation.Yes)
+		initGit, _ := gitPrompt.RunPrompt()
+		if initGit {
+			err := utils.InitGit(projectName)
+			if err != nil {
+				fmt.Println(" - Failed to initialize Git repository.")
+			} else {
+				fmt.Println(" + Git repository initialized.")
+			}
+		}
+
+		if selectedTemplate.TSConfig_prompt {
+			tscPrompt := confirmation.New(" ? Would you like to add TypeScript configuration file", confirmation.No)
+			tsc, err := tscPrompt.RunPrompt()
+			if err != nil && tsc {
+				_, err := grab.Get(projectName, "https://raw.githubusercontent.com/grammyjs/cli/main/configs/tsconfig.json")
+				if err != nil {
+					fmt.Println(" - Skipping... Failed to add a tsconfig.json.")
+				}
+			}
+		}
+
+		if selectedTemplate.Docker_prompt {
+			dockerPrompt := confirmation.New(" ? Add Docker related files", confirmation.No)
+			docker, err := dockerPrompt.RunPrompt()
+			if err == nil && docker {
+				err := files.AddDockerFiles(platform, projectName)
+				if err != nil {
+					fmt.Println(" - Failed to add Docker files.")
+				}
+			}
+		}
+
+		var action string
+
+		switch platform {
+		case "Deno":
+			action = "Cache"
+		case "Node":
+			action = "Install"
+		case "Other":
+			fmt.Printf(footer, projectName)
+			return
+		}
+
+		pkgInstallPrompt := confirmation.New(fmt.Sprintf(" ? %s dependencies", action), confirmation.Yes)
+		installPkgs, err := pkgInstallPrompt.RunPrompt()
+
+		if err != nil {
+			os.Exit(1)
+		}
+
+		if installPkgs {
```

Comment:
Same point like for promts, it should be splitted to internal, not there. Same for `tsconfig`, `Docker` setup.